compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3680,7 +3687,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 _ => None,
             };
             if !trait_missing_method {
-                err.help(if param_type.is_some() {
+                err.span_help(span, if param_type.is_some() {
```

Comment:
Please revert this. I don't think this improves the quality of the error message.

Reply:
> Please revert this. I don't think this improves the quality of the error message.

Done. Thank you.